src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/projectcreation/model/ProjectCreationModel.kt

```diff
@@ -21,50 +22,81 @@ class ProjectCreationModel {
+    var collectionStore: ArrayList<List<Collection>> = ArrayList()
+    var creationDepth: Int = 1
+    var bookDepthReached = false
 
     init {
         creationUseCase.getAllLanguages()
                 .observeOnFx()
                 .subscribe { retrieved ->
                     languages.setAll(retrieved)
                 }
+    }
 
+    fun getSourceRepos() {
         creationUseCase.getSourceRepos()
                 .observeOnFx()
                 .subscribe { retrieved ->
-                    resources.setAll(retrieved)
+                    collectionStore.add(retrieved.filter {
+                        it.resourceContainer?.language == sourceLanguageProperty
+                    })
+                    collectionList.setAll(collectionStore[collectionStore.size - 1])
                 }
     }
 
-    fun getResourceChildren() {
-        creationUseCase.getResourceChildren(selectedResource)
-                .observeOnFx()
-                .doOnSuccess {
-                    anthologyList.setAll(it.sortedBy { it.sort })
-                }
-                .subscribe()
+    fun checkLevel(selectedCollection: Collection, workspace: Workspace) {
+        if (collectionList[0].labelKey == "book") {
+            createProject(selectedCollection)
+            workspace.dock<ProjectHomeView>()
+        } else {
+            getResourceChildren(selectedCollection)
+        }
     }
 
-    fun getBooks() {
-        creationUseCase.getResourceChildren(selectedAnthology)
+
+    private fun getResourceChildren(parentCollection: Collection) {
+        creationUseCase.getResourceChildren(parentCollection)
                 .observeOnFx()
                 .doOnSuccess {
-                    bookList.setAll(it.sortedBy { it.sort })
+                    collectionStore.add(it)
+                    collectionList.setAll(collectionStore[collectionStore.size - 1].sortedBy { it.sort })
+                    getDepth(parentCollection)
                 }
                 .subscribe()
     }
 
-    fun createProject() {
+    fun getPreviousCollections(projectWizard: Wizard) {
+        if (collectionStore.size > 1) {
+            collectionStore.removeAt(collectionStore.size - 1)
+            collectionList.setAll(collectionStore[collectionStore.size - 1].sortedBy { it.sort })
+        } else {
+            projectWizard.back()
+        }
+    }
+
+    fun getDepth(selectedCollection: Collection) {
+        if (creationDepth == 1) {
+                creationUseCase.getResourceChildren(selectedCollection)
+                        .observeOnFx()
+                        .subscribe { retrieved ->
+                            if (retrieved[0].labelKey == "book") {
+                                bookDepthReached = true
+                            } else {
+                                creationDepth++
+                                print(creationDepth)
+                                getDepth(retrieved[0])
```

Comment:
Does this need to be recursive or can it just check whether the next layer is a book?